src/lua/functions/core/game/bank_functions.cpp

```diff
@@ -96,7 +96,12 @@ int BankFunctions::luaBankWithdraw(lua_State* L) {
 	auto player = getPlayer(L, 1);
 	uint64_t amount = getNumber<uint64_t>(L, 2);
 	if (lua_gettop(L) == 2) {
-		auto bank = std::make_unique<Bank>(player);
+		if (!player) {
+			return 1;
+		}
+
+		const auto &bankablePlayer = std::shared_ptr<Bankable>(player);
```

Comment:
The reason I didn't make `Bankable` be an autoptr originally was it that I was afraid it could free the player here, now guild is an autoptr I understand we need something different, but this might have to be `std::weak_ptr` instead to avoid destroying the player too early